iothub/device/samples/PnpDeviceSamples/TemperatureController/Program.cs

```diff
@@ -0,0 +1,327 @@
+
+            await RunSampleAsync();
+        }
+
+        private static async Task RunSampleAsync()
+        {
+            // This sample follows the following workflow:
+            // -> Initialize device client instance.
+            // -> Set handler to receive "reboot" command - root interface.
+            // -> Set handler to receive "getMaxMinReport" command - on "Thermostat" components.
+            // -> Set handler to receive "targetTemperature" property updates from service - on "Thermostat" components.
+            // -> Send initial device info - "workingSet" over telemetry, "serialNumber" over reported property update - root interface.
+            // -> Periodically send "temperature" over telemetry - on "Thermostat" components.
+            // -> Send "maxTempSinceLastReboot" over property update, when a new max temperature is set - on "Thermostat" components.
+
+            s_logger.LogDebug($"Initialize the device client.");
+            InitializeDeviceClientAsync();
+
+            s_logger.LogDebug($"Set handler for \"reboot\" command.");
+            await s_deviceClient.SetMethodHandlerAsync("reboot", HandleRebootCommandAsync, s_deviceClient);
+
+            s_logger.LogDebug($"Set handler for \"getMaxMinReport\" command.");
+            await s_deviceClient.SetMethodHandlerAsync("thermostat1*getMaxMinReport", HandleMaxMinReportCommandAsync, Thermostat1);
+            await s_deviceClient.SetMethodHandlerAsync("thermostat2*getMaxMinReport", HandleMaxMinReportCommandAsync, Thermostat2);
+
+            s_logger.LogDebug($"Set handler to receive \"targetTemperature\" updates.");
+            await s_deviceClient.SetDesiredPropertyUpdateCallbackAsync(SetDesiredPropertyUpdateCallbackAsync, null);
+            s_desiredPropertyUpdateCallbacks.Add(Thermostat1, TargetTemperatureUpdateCallbackAsync);
+            s_desiredPropertyUpdateCallbacks.Add(Thermostat2, TargetTemperatureUpdateCallbackAsync);
+
+            await SendDeviceMemoryAsync();
+            await SendDeviceSerialNumberAsync();
+
+            bool temperatureReset = true;
+            s_maxTemp[Thermostat1] = 0d;
+            s_maxTemp[Thermostat2] = 0d;
+
+            await Task.Run(async () =>
+            {
+                while (true)
+                {
+                    if (temperatureReset)
+                    {
+                        // Generate a random value between 5.0°C and 45.0°C for the current temperature reading for each "Thermostat" component.
+                        s_temperature[Thermostat1] = Math.Round(s_random.NextDouble() * 40.0 + 5.0, 1);
+                        s_temperature[Thermostat2] = Math.Round(s_random.NextDouble() * 40.0 + 5.0, 1);
+                    }
+
+                    await SendTemperatureAsync(Thermostat1);
+                    await SendTemperatureAsync(Thermostat2);
+
+                    temperatureReset = s_temperature[Thermostat1] == 0 && s_temperature[Thermostat2] == 0;
```

Comment:
I am not following the logic behind your reset. You want to reset when temp of each thermostat turns to 0 and from your logic on line 107, 108 this value will always be between 5 and 40 ?

Reply:
Yes, I should have added a comment here: this is an open item right now, the `Thermostat` interface does not have a "reboot" command, but has a property for holding max temp since last reset: https://github.com/Azure/azure-iot-sdk-csharp/pull/1423#discussion_r449310913

Based on his update, I'll remove the reset logic from `Thermostat` sample,. This logic makes sence for the `TemperatureController` sample, which does have a "reboot" command.